File: PYTHON/Shortest Word.py
"""
Simple, given a string of words, return the length of the shortest word(s).

String will never be empty and you do not need to account for different data types.
"""
def find_short(s):
    words = s.split(' ')
    shortest_length = len(words[0])
    
    for word in words:
        current_length = len(word)
        
        if current_length < shortest_length:
            shortest_length = current_length
    
    return shortest_length

# ALTERNATIVES
# return len(min(s.split(' '), key=len))
# return min(len(x) for x in s.split())

File: Codewars/2024/C++/7kyu/shortest_word.cpp
#include <string>
#include <sstream>
#include <limits>

/**
 * @brief Finds the length of the shortest word in a given string.
 * 
 * @details This function takes a string of words and returns the length
 *          of the shortest word(s) in it. The string will never be empty,
 *          and there's no need to account for different data types.
 * 
 * @param str A string containing one or more words.
 * @return int The length of the shortest word in the string.
 */
int find_short(std::string str)
{
    std::istringstream iss(str);
    std::string word;
    int shortest = std::numeric_limits<int>::max();
    
    while (iss >> word) {
        int length = word.length();
        if (length < shortest) {
            shortest = length;
        }
    }
    
    return shortest;
}
